src/Algebra/Module/Construct/Nagata.agda

```diff
@@ -0,0 +1,170 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The non-commutative analogue of Nagata's construction
+-- of the "idealization of a module", defined here
+-- on R-R-*bi*modules M over a ring R.
+--
+-- Elements of R ⋉ M are pairs |R| × |M|
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Algebra.Module.Construct.Nagata where
+
+open import Algebra.Bundles
+open import Algebra.Core
+import Algebra.Consequences.Setoid as Consequences
+import Algebra.Definitions as Definitions
+open import Algebra.Module.Bundles
+open import Algebra.Module.Core
+open import Algebra.Module.Definitions
+import Algebra.Module.Construct.DirectProduct as DirectProduct
+import Algebra.Module.Construct.TensorUnit as TensorUnit
+open import Algebra.Structures
+open import Data.Product using (_×_; _,_; proj₁; proj₂)
+open import Data.Product.Relation.Binary.Pointwise.NonDependent
+open import Level using (Level; _⊔_)
+open import Relation.Binary using (Rel; Setoid; IsEquivalence)
+import Relation.Binary.Reasoning.Setoid as ≈-Reasoning
+
+private
+  variable
+    m ℓm r ℓr : Level
+
+------------------------------------------------------------------------
+-- Definitions
+
+module Nagata (ring : Ring r ℓr) (bimodule : Bimodule ring ring m ℓm) where
+
+  open Ring ring
```

Comment:
Instead of all this renaming I think a nicer pattern everywhere would be
```agda
open private module R = Ring ring 
  renaming (Carrier to R)
```
then you can simply use `R._+_` rather than `_+ᴿ_`  (which is easier to type anyway). This is consistent with how we resolve clashes with overloaded operators in `Data.Rational` etc. This point should probably be added to the style guide....

Reply:
Yes, this does (as with other review comments you've made about my use of modules in eg #1962 ; sorry to be such a slow learner, but this Nagata code is now quite old, and I hadn't really brought it up to date for this PR, and should have before committing... )

Reply:
> Instead of all this renaming I think a nicer pattern everywhere would be
> 
> ```agda
> open private module R = Ring ring 
>   renaming (Carrier to R)
> ```

Hmmm... this doesn't seem quite to work. What I can get working is 
```agda
private module R = Ring ring
open R renaming (...)
```
it doesn't seem (v2.6.4) to like collapsing the `private` definition with the `open` declaration?